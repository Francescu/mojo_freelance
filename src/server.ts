import type { Request, Response } from "express";

import express from "express";

import getSkills from "./freelancers/get-skills";

const server = express();

const notFound = (req: Request, res: Response) => {
  res.status(404).json({});
};

server.get("/freelancer/:id/skills", getSkills);
server.use(notFound);

export default server;
