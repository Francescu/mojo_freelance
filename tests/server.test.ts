import server from "../src/server";

import request from "supertest";

console.error = jest.fn();

describe("Test Freelancer API", () => {
  test("Expect 404 not found on invalid route", async () => {
    const result = await request(server).get("/");
    expect(result.body).toEqual({});
    expect(result.statusCode).toEqual(404);
  });

  describe("GET /freelancer/:id/skills", () => {
    test("Expect to fail if id is invalid", async () => {
      const result = await request(server).get("/freelancer/invalidID/skills");
      expect(result.text).toEqual("Server error.");
      expect(result.statusCode).toEqual(500);
    });

    test("Expect to return valid skill experience duration on success", async () => {
      const result = await request(server).get("/freelancer/42/skills");
      expect(result.statusCode).toEqual(200);
      expect(result.body).toMatchSnapshot();
    });
  });
});
