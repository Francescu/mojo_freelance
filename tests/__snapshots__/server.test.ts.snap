// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Freelancer API GET /freelancer/:id/skills Expect to return valid skill experience duration on success 1`] = `
Object {
  "freelance": Object {
    "computedSkills": Array [
      Object {
        "durationInMonths": 28,
        "id": 241,
        "name": "React",
      },
      Object {
        "durationInMonths": 28,
        "id": 270,
        "name": "Node.js",
      },
      Object {
        "durationInMonths": 60,
        "id": 370,
        "name": "Javascript",
      },
      Object {
        "durationInMonths": 32,
        "id": 470,
        "name": "MySQL",
      },
      Object {
        "durationInMonths": 40,
        "id": 400,
        "name": "Java",
      },
    ],
    "id": 42,
  },
}
`;
